server.js

```diff
@@ -1,23 +1,110 @@
 const express = require("express");
-const cors = require("cors");
+const cors = require('cors')
+var bodyParser = require('body-parser');
 
 const app = express();
-
-app.use(cors());
-
+const port = 3001;
+app.use(cors())
+app.use(bodyParser.urlencoded({ extended: true })); 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
-};
+  timeSent: new Date(2020, 11, 17, 3, 24, 0)
+}
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+const messages = [welcomeMessage]
+
+
+app.get('/', function(request, response) {
+  response.sendFile(__dirname + '/index.html');
+});
+//reading all messages
+app.get('/messages', function(request, response) {
+  response.json(messages);
+});
+
+//creating new message
+app.post('/messages',  (req, res) => {
+  let message={
+    id:messages.length,
```

Comment:
There is a problem with setting the ID as the length of messages.
For example, if you have 3 messages with IDs 0,1,2 and you delete message 1, then add another message, this new message will have an ID of 2. This will give you 2 messages with an ID of 2, and IDs have to be unique.